### DISCUSSION TITLE
KFP v2 add_port

### DISCUSSION THREAD

**Discussion by 482170765** [kubeflow/pipelines#11188] [General]
kfp == 2.8.0

kfp-kubernetes == 1.2.0

@dsl.component()
def server():
    ....

@dsl.pipeline()
def my_pipeline():
    server_task=server()
    server_task.add_port(V1ContainerPort(name='my-port', container_port=8080))
---> server_task.add_port(V1ContainerPort(name='my-port', container_port=8080))

AttributeError: 'PipelineTask' object has no attribute 'add_port'

In kfpv1 the code can work, but in kfpv2 'add_port' had been removed. Is there any method to add port on task ?

**Comment by rimolive** [comment]
Can you describe what you want to do with exposing ports in the pipeline step pods?

**Reply by sefgsefg** [reply]
Do you know how to replace it? I got same problem about replace "add_port" funtion.

**Comment by 482170765** [comment]
@rimolive the purpose of add_port is as follows: we wish to allow different pods to communicate with each other.  Each pod is loaded with different programs which needs to exchange data for federal learning applications.
